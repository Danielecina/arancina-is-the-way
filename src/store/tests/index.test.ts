import {chrome} from 'jest-chrome'

import {promiseStore} from '..'

describe('store', () => {
  test('expect to return correct store', async () => {
    chrome.storage.sync.get.mockImplementation((key, resolveCallback) => {
      resolveCallback({})
    })

    const store = await promiseStore()
    expect(store.getState()).toMatchSnapshot()
  })

  test('expect to correct saving data to chrome storage', async () => {
    chrome.storage.sync.get.mockImplementation((key, resolveCallback) => {
      resolveCallback({substitution: {watchMode: true}})
    })

    const dataSendToChromeStorage = jest.fn()
    chrome.storage.sync.set.mockImplementation(dataSendToChromeStorage)

    const store = await promiseStore()
    store.dispatch({type: 'TOGGLE_WATCH_MODE', payload: {watchMode: false}})

    expect(store.getState()).toEqual({
      ...store.getState(),
      substitution: {watchMode: false}
    })
    expect(dataSendToChromeStorage).toHaveBeenCalledWith({
      ...store.getState(),
      substitution: {watchMode: false}
    })
  })
})
