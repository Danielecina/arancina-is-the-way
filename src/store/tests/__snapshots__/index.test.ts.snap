// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store expect to return correct store 1`] = `
Object {
  "language": Object {
    "locale": "en",
    "selected": "browser",
  },
  "substitution": Object {
    "watchMode": false,
  },
}
`;
